src/ucm/util/reloc.h

```diff
@@ -55,12 +55,32 @@ static UCS_F_MAYBE_UNUSED
 void* ucm_reloc_get_orig(const char *symbol, void *replacement)
 {
     const char *error;
-    void *func_ptr;
+    void *func_ptr = NULL;
+    int ret;
+    void *dl;
+    Dl_info info;
+    int flags = RTLD_LOCAL | RTLD_NODELETE | RTLD_LAZY;
+
+    (void)dlerror();
+    ret = dladdr((void*)ucm_reloc_get_orig, &info);
+    if (ret == 0) {
+        ucm_warn("could not find address of current library: %s", dlerror());
+        return NULL;
+    }
+
+    (void)dlerror();
+    dl = dlopen(info.dli_fname, flags);
+    if (dl != NULL) {
+        (void)dlerror();
+        func_ptr = dlsym(dl, symbol);
```

Comment:
this changes the behavior: before the change the symbol was searched in the caller library (that could be libucm_cuda.so for example) and now it's always searched in libucm.so

Reply:
is it fixing a failure that happens with RISC-V? what was that change needed?
I'm worried it may break other use cases..

Reply:
I understand - since this was specifically seen in with the UCC gtest (where RTLD_NEXT fell through to patching the PLT) i'm going to propose moving this to a separate PR/GitHub issue where we can discuss the behaviour (i'd also like to see if that fallthough to patching the PLT also occurs with the UCC gtest on x86-64.)

Reply:
can we remove it from this PR then?